test/thirdparty.test.ts

```diff
@@ -0,0 +1,227 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { deepStrictEqual, strictEqual } from "assert";
+import { describe, test } from "mocha";
+import { getAccounts } from "@tableland/local";
+import {
+  D1Orm,
+  DataTypes,
+  Model,
+  GenerateQuery,
+  QueryType,
+  type Infer,
+} from "d1-orm";
+import sql, { FormatConfig } from "@databases/sql";
+import { escapeSQLiteIdentifier } from "@databases/escape-identifier";
+import { NonceManager } from "@ethersproject/experimental";
+import { getDefaultProvider, overrideDefaults } from "../src/helpers/index.js";
+import { Database } from "../src/index.js";
+
+overrideDefaults("localhost", {
+  contractAddress: "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
+});
+
+describe("thirdparty", function () {
+  // Note that we're using the second account here
+  const [, wallet] = getAccounts();
+  const provider = getDefaultProvider("http://127.0.0.1:8545");
+  // const signer = wallet.connect(provider);
+  const baseSigner = wallet.connect(provider);
+  // Also demonstrates the nonce manager usage
+  const signer = new NonceManager(baseSigner);
+  const db = new Database({ signer });
+
+  describe("d1-orm", function () {
+    const orm = new D1Orm(db);
+
+    // We'll define our core model up here and use it in tests below
+    const users = new Model(
+      {
+        D1Orm: orm,
+        tableName: "users",
+        primaryKeys: "id",
+        uniqueKeys: [["email"]],
+      },
+      {
+        id: {
+          type: DataTypes.INTEGER,
+          notNull: true,
+        },
+        name: {
+          type: DataTypes.STRING,
+          notNull: true,
+          // See https://github.com/Interactions-as-a-Service/d1-orm/issues/60
+          // defaultValue: "John Doe",
+        },
+        email: {
+          type: DataTypes.STRING,
+        },
+      }
+    );
+    type User = Infer<typeof users>;
+
+    this.beforeAll(async function () {
+      this.timeout("10s");
+
+      const create = await users.CreateTable({
```

Comment:
Even creating tables works?! Rad. In this case though, we need a slight "hack" to update the tableName, which is marked as immutable (though can be manipulated if we force it).